Update the talk about release version to the v1 world

oxinabox opened the pull request:
This is my attempt.
I don't think it is great
But in a generalization of [Cunningham's Law:](https://meta.wikimedia.org/wiki/Cunningham%27s_Law)
the fastest way to get this fixed is for me to try and fix it and fail.

Since github does not do multiline suggestions yet,
feel free to just push more commits onto this branch.

oxinabox committed:
Try and update the talk about release version to the 1.0 world

ViralBShah commented:
@JeffBezanson @StefanKarpinski It would be great for you to review this.

c42f commented on doc/src/manual/faq.md:
```diff
@@ -859,17 +859,33 @@ Modifying OpenBLAS settings or compiling Julia with a different BLAS library, eg
+A new minor release of Julia corresponding to a new Stable version is made approximately every 4 months.
+After a few weeks of testing as a release candidate.
+Unlike the LTS version the a Stable version will not normally recieve bugfixes after another Stable version of Julia has been released.
+However, upgrading to the next Stable release will always be possible: as each release of Julia v1.x
```
grammar: "possible: as" -> "possible:"

c42f reviewed (commented):
FWIW this looks good to me.

ViralBShah commented:
I think there is a way to submit changes that can be accepted. Can you do that?

oxinabox commented on doc/src/manual/faq.md:
```diff
@@ -880,5 +896,7 @@ Note that not all versions of Julia are available for all platforms.
+Deprecated functions are removed after the subsequent major release.
+At time of writing, there is no concrete timeline for the next major release of julia.
+For example, functions marked as
+deprecated in the 1.1 release will not be available starting with the 2.0 release.
```
I thought we were doing silent deprecation?

c42f committed:
Tweak FAQ

* Fix some grammar
* Remove mention of deprecated functions, as it seems there will be no hard deprecations during the 1.x release cycle. When 2.0 comes around the policy is likely to be different from the 0.n -> 0.(n+1) releases.

c42f commented:
@ViralBShah it seems it's possible to edit the file directly on this branch. I've done that for now but perhaps there's a better way.

I removed the section about deprecated functions for now because I imaging the deprecation policy will be completely revisited when we eventually get to 2.0 and may not resemble the 0.x cycle at all. But perhaps there's an official position on that which I don't know about?

KristofferC commented on doc/src/manual/faq.md:
```diff
@@ -859,26 +859,36 @@ Modifying OpenBLAS settings or compiling Julia with a different BLAS library, eg
+The Stable version of Julia is the latest released version of Julia,
+this is the version most people will want to run.
+It has the latest features, including improved performance.
+The Stable version of Julia will always be versioned according to SemVer as v1.x.y.
```
always -> currently?

KristofferC commented on doc/src/manual/faq.md:
```diff
@@ -859,26 +859,36 @@ Modifying OpenBLAS settings or compiling Julia with a different BLAS library, eg
+However, upgrading to the next Stable release will always be possible as each release of Julia v1.x
+will continue to run code written for earlier versions.
+
+You may prefer the LTS (Long Term Support) version of Julia if you are looking for a very stable code base.
```
The acronym LTS is already mentioned above so should have the full name there.

KristofferC commented on doc/src/manual/faq.md:
```diff
@@ -859,26 +859,36 @@ Modifying OpenBLAS settings or compiling Julia with a different BLAS library, eg
+The Stable version of Julia will always be versioned according to SemVer as v1.x.y.
+A new minor release of Julia corresponding to a new Stable version is made approximately every 4 months after a few weeks of testing as a release candidate.
+Unlike the LTS version the a Stable version will not normally recieve bugfixes after another Stable version of Julia has been released.
+However, upgrading to the next Stable release will always be possible as each release of Julia v1.x
```
2.0 will be a breaking stable release so this isn't really true.

KristofferC commented on doc/src/manual/faq.md:
```diff
@@ -859,26 +859,36 @@ Modifying OpenBLAS settings or compiling Julia with a different BLAS library, eg
+will continue to run code written for earlier versions.
+
+You may prefer the LTS (Long Term Support) version of Julia if you are looking for a very stable code base.
+The LTS version of Julia will always be versioned according to SemVer as v1.0.x.
```
I don't think this is true. At some point we will make a new v1.x the LTS.

KristofferC commented on doc/src/manual/faq.md:
```diff
@@ -859,26 +859,36 @@ Modifying OpenBLAS settings or compiling Julia with a different BLAS library, eg
+You may prefer the LTS (Long Term Support) version of Julia if you are looking for a very stable code base.
+The LTS version of Julia will always be versioned according to SemVer as v1.0.x.
+It will continue to recieve bugfixes for the lifetime of Julia version 1, but will not recieve any new features.
+As a package developer, you may prefer to develop for the LTS version, to maximize the number of users who can use your package.
```
I don't think this is such a good advice. Only people who are very conservative and want their existing code to run should stay on LTS and then you are likely not adding a bunch of new packages. Developing for the latest stable release should imo be the default.

oxinabox replied:
I present both alternatives. and the pros and cons of each.
It says *may*. not *should*.

KristofferC commented on doc/src/manual/faq.md:
```diff
@@ -859,26 +859,36 @@ Modifying OpenBLAS settings or compiling Julia with a different BLAS library, eg
+The LTS version of Julia will always be versioned according to SemVer as v1.0.x.
+It will continue to recieve bugfixes for the lifetime of Julia version 1, but will not recieve any new features.
+As a package developer, you may prefer to develop for the LTS version, to maximize the number of users who can use your package.
+Code written for v1.0 will continue to work for all future LTS and Stable versions.
```
Already said on line 870-871?

oxinabox replied:
870-871 was in reference to the Stable versions

KristofferC commented on doc/src/manual/faq.md:
```diff
@@ -859,26 +859,36 @@ Modifying OpenBLAS settings or compiling Julia with a different BLAS library, eg
+Code written for v1.0 will continue to work for all future LTS and Stable versions.
+In general even if targetting the LTS, one can develop and run code in the latest Stable version,
+to take advantage of the improved performance; so long as one is careful not to use any of the new features (such as added library functions).
+Patch releases to the LTS version generally occur approximately at the same time as releases to the Stable version.
```
I don't think this is necessarily the case.

oxinabox replied:
it does say _generally_

KristofferC commented on doc/src/manual/faq.md:
```diff
@@ -859,26 +859,36 @@ Modifying OpenBLAS settings or compiling Julia with a different BLAS library, eg
+As the name implies, releases to the nightly version are made every night.
+In general nightly released are fairly safe to use.
+However, they may have regressions and or issues that have not yet been caught.
+You may wish to test against the nightly version to ensure that such regressions that affect your use case are caught before a release is made.
```
I think this sounds a bit too alarming. There is a decently long RC process where there is time to test for regressions.

oxinabox replied:
again it says _may_.
We have had breaking changed caught (and recently) thanks to people doing this,
I don't see any reason for them to stop.

StefanKarpinski committed:
release FAQ: clarify that neither stable=1.x nor LTS=1.0.x are forever

- At some point the stable series will be 2.x, 3.x, etc.
- At some point the LTS will no longer be 1.0.x, it could be 1.6.x

StefanKarpinski committed:
release FAQ: a couple of tweaks

fredrikekre committed:
Update doc/src/manual/faq.md

Co-Authored-By: oxinabox <oxinabox@ucc.asn.au>

oxinabox commented:
bump
